File: isi-backend/src/main/java/com/guuri11/isi/domain/fav/Fav.java
package com.guuri11.isi.domain.fav;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.UUID;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
public class Fav {

  @Id
  @GeneratedValue(strategy = GenerationType.UUID)
  private UUID id;
  // TODO: handle exception in create service
  @Column(name = "name", unique = true)
  private String name;
  private LocalDateTime createdAt;
  private LocalDateTime updatedAt;

}

File: isi-multiplatform/composeApp/src/commonMain/kotlin/domain/entity/Fav.kt
package domain.entity

import kotlinx.datetime.LocalDateTime
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import utils.LocalDateTimeSerializer

data class Fav(
    val id: String,
    val name: String,
    @Serializable(with = LocalDateTimeSerializer::class) @SerialName("createdAt") val createdAt: LocalDateTime,
    @Serializable(with = LocalDateTimeSerializer::class) @SerialName("updateAt") val updatedAt: LocalDateTime
)
